### ISSUE TITLE
Variable in Smart contract in ganache is not updated

### ISSUE THREAD

**Issue by pstrotmann** [trufflesuite/truffle#4659]
Truffle Ticket #702341 [Pending] Support Ticket sent from peter@strotmann.org in HubSpot - Truffle

All you need to reproduce the Issue is contained in the repo:

https://github.com/pstrotmann/SimpleStorage

I would expect storedData = 17 in my smart contract after running simple_storage.js because the assert after calling

```
// set simpleStorage
   await simpleStorageInstance.set.sendTransaction(17, { from: accounts[0] });
   simpleStorageInstance.set.sendTransaction(17, {from:accounts[0]});

// Get stored data
    const storedData = await simpleStorageInstance.get();

    assert.equal(storedData, 17, "The value 17 was not stored.");
```

indicates that 17 is stored in storedData.

Therefore I would expect that the 17 is stored persistent in SimpleStorage contract. But as You see in ganache there is 0.

The 3 steps I performed (see hardcopies in obove repo) were:

1) truffle deploy (to deploy the contract SimpleStorage, as you see in ganacheTest/console/truffleDeploy.png)

2) truffle test ./test/simple_storage.js (to test setting storedData = 17 in contract SimpleStorage, as you see in ganacheTest/console/truffleTestSimpleStorageJs.png)

3) truffle test ./test/simple_storage.sol (to test setting storedData = 18 in contract SimpleStorage, as you see in ganacheTest/console/truffleTestSimpleStorageSol.png)

As you see from the hardcopies deploy and the tests succeeded either.

You also see from the harcopies named worspace... what happened due to deploying and testing on the ganache blockchain on the account, the blocks, transactions and the contract after each of the above steps. 

**Without the update facility of smart contracts in truffle/ganache testing contracts does not make real sense making truffle/ganache quite worthless.**

**Comment by pstrotmann** [comment]
an option --norollback in the test command would meet my needs completely

**Comment by eggplantzzz** [comment]
In the meantime, tell me what you want to happen in general and maybe I can give you a workaround. Like, what are your needs?

**Comment by pstrotmann** [comment]
What do you mean with meantime ? Is it the time until test --norollback is implemented ? If that time is under 3 months there is no need for a workaround.

**Comment by eggplantzzz** [comment]
I do not know if/when we will  implement that feature but we will put it in the backlog for consideration. So to be clear, your needs are that you want to be able to programmatically test your truffle project and have the test network state never revert. Is that correct? The testing flow is the only time where such state reversions happen.

**Comment by alizdd** [comment]
I agree with @pstrotmann , I was confused and lost time when I see no changes in ganache interface, with norollback option, that would be better test experience imo

**Comment by amva13** [comment]
to be clear -- is this also the reason the state seems to rollback during webapp testing with react. say I have a setter and a getter. I call the setter first and everything seems to work and console.log reflects the change in ganache. I call the getter but the variable is now back to its initial state. as @eggplantzzz mentioned, doing everything from truffle develop works perfectly. from react webapp, rollback seems to happen after an individual contract function call instead. is a react function call treated as an individual test here ? if so, I think a no rollback option would be useful for webapp testing. I agree with @eggplantzzz rolling back after an individual test is desired behavior for automated testing suites.
